src/controller/python/chip/internal/CommissionerImpl.cpp

```diff
@@ -156,22 +165,51 @@ class ScriptDevicePairingDelegate final : public chip::Controller::DevicePairing
 
     /// Delegate is set during request callbacks
     chip::RendezvousDeviceCredentialsDelegate * mCredentialsDelegate = nullptr;
-
-    // Copy of wifi credentials, to allow them to be used by callbacks
-    std::string mWifiSsid;
-    std::string mWifiPassword;
 };
 
 ServerStorageDelegate gServerStorage;
 ScriptDevicePairingDelegate gPairingDelegate;
 
 } // namespace
 
-extern "C" void pychip_internal_PairingDelegate_SetNetworkCredentials(const char * ssid, const char * password)
+extern "C" void pychip_internal_PairingDelegate_SetWifiCredentials(const char * ssid, const char * password)
 {
     chip::python::ChipMainThreadScheduleAndWait([&]() { gPairingDelegate.SetWifiCredentials(ssid, password); });
 }
 
+extern "C" CHIP_ERROR pychip_internal_PairingDelegate_SetThreadCredentials(const void * data, uint32_t length)
+{
+
+    // Openthread is OPAQUE by the spec, however current CHIP stack does not have any
```

Comment:
This format is not the specified format. They have roughly the same contents, but this isn't the representation of that data that we're required to use.

We're not meant to do any validation of the dataset.

Reply:
Are you suggesting to remove this check completely?

My plan was to have it as a TODO to remove once our entire stack treats it as opaque. Until then, our stack requires a specific structure usage, so we have no choice but to memcpy the data (or cast content), which seems to likely just crash on invalid data. Figure a basic sanity check is easier to debug than a crash.

Totally agree on long term - this should move around opaque data and not interpret it.